subprojects/docs/src/snippets/dependencyManagement/customizingResolution-attributeSubstitutionRule/kotlin/consumer/build.gradle.kts

```diff
@@ -30,8 +30,10 @@ configurations.all {
 // end::substitution_rule_alternative[]
 
 tasks.register("resolve") {
```

Comment:
This fix eagerly accesses the `configurations` object during the configuration phase, storing the result as a value that is configuration-cache friendly, instead of during the execution phase, where access to `configurations` would not be allowed. See: https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:disallowed_types

Reply:
> This fix eagerly accesses the `configurations` object during the configuration phase, storing the result as a value that is configuration-cache friendly, instead of during the execution phase, where access to `configurations` would not be allowed. 

Note that it is important to specify the type of the variable, even though we're storing the actual `Configuration` object in it. Having a simpler type signals the configuration cache that there's no need to preserve a whole complex configuration object, but it can be stored in a simplified form to be restored as a `FileCollection` later.